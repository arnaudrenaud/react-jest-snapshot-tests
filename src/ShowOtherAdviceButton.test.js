import React from 'react';
import { shallow } from 'enzyme';
import ShowOtherAdviceButton from "./ShowOtherAdviceButton";

describe('ShowOtherAdviceButton', () => {
  const handleClickMock = jest.fn();

  it('renders correctly', () => {
    const wrapper = shallow(<ShowOtherAdviceButton onClick={handleClickMock} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when button clicked', () => {
    it('calls prop onClick', () => {
      const wrapper = shallow(<ShowOtherAdviceButton onClick={handleClickMock} />);
      wrapper.find('button').simulate('click');
      expect(handleClickMock).toHaveBeenCalledTimes(1);
    });
  });
});