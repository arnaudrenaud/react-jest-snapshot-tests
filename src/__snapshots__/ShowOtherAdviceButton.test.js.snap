// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowOtherAdviceButton renders correctly 1`] = `
<button
  className="Advice-other-advice-button"
  onClick={[MockFunction]}
>
  Show other advice
</button>
`;
